src/components/secrets/RevisionField/RevisionField.tsx

```diff
@@ -0,0 +1,41 @@
+import type { PropsWithSpread, SelectProps } from "@canonical/react-components";
+import { Select } from "@canonical/react-components";
+
+import type { FormikFieldProps } from "components/FormikField";
+import FormikField from "components/FormikField";
+import { getSecretByURI } from "store/juju/selectors";
+import { useAppSelector } from "store/store";
+
+export enum Label {
+  REVISION = "Revision",
+}
+
+type Props = PropsWithSpread<
+  {
+    modelUUID: string;
+    secretURI: string;
+  },
+  Partial<FormikFieldProps> & Partial<SelectProps>
+>;
+
+const RevisionField = ({ modelUUID, secretURI, ...props }: Props) => {
```

Comment:
I've added this component as the same thing needs to be done in the remove secret form.